### ISSUE TITLE
Support for hash nested under an array

### ISSUE THREAD

**Issue by spudly** [defunctzombie/form-serialize#18]
I expected to get an array from a form like this, but got an object instead.

```html
<form>
  <input name='people[0][name]' value='fred' />
  <input name='people[0][age]' value='12' />

  <input name='people[1][name]' value='bob' />
  <input name='people[1][age]' value='14' />

  <input name='people[2][name]' value='bubba' />
  <input name='people[2][age]' value='15' />
</form>
```

Desired output:
```javascript
{
  "people": [
    {"name": "fred", "age": "12"},
    {"name": "bob", "age": "14"},
    {"name": "bubba", "age": "15"}
  ]
}
```

Actual output:
```javascript
{
  "people": { // <-- people is not an array. darn.
    "0": {"name": "fred", "age": "12"},
    "1": {"name": "bob", "age": "14"},
    "2": {"name": "bubba", "age": "15"}
  }
}
```

I think that this expectation is consistent with what other libraries do, since I've used this form structure in the past. Any chance this form structure could be supported?

**Comment by defunctzombie** [comment]
Have you tried removing the `0, 1, 2` from between the `[]` and just leaving `[]`. I think the lib detects that and will make the array.

**Comment by spudly** [comment]
Yes, I tried that, but here's the result...

```html
<form>
  <input name="people[][name]" value="fred">
  <input name="people[][age]" value="12">

  <input name="people[][name]" value="bob">
  <input name="people[][age]" value="14">

  <input name="people[][name]" value="bubba">
  <input name="people[][age]" value="15">
</form>
```

```json
{
  "people": {
    "][name": ["fred", "bob", "bubba"],
    "][age": ["12", "14", "15"]
  }
}
```

**Comment by spudly** [comment]
For anyone else who may need this, I've temporarily worked around my issue by using this module to parse the form into a querystring and the [qs module](https://github.com/hapijs/qs)  to parse the querystring into an object.

**Comment by defunctzombie** [comment]
So the issue is with using [] followed by another [] it seems.

What if you omit the first []? Does the library figure out you have
multiple?

**Comment by spudly** [comment]
I assume you mean like this:

```html
<form>
  <input name="people[name]" value="fred">
  <input name="people[age]" value="12">

  <input name="people[name]" value="bob">
  <input name="people[age]" value="14">

  <input name="people[name]" value="bubba">
  <input name="people[age]" value="15">
</form>
```

That gives the following output:

```json

```

By the way, I created a test for this on runnable.com to make it easier to test out different variations: http://runnable.com/VVDyUCD_d_hq0H8h/form-serialize-issue-18

**Comment by defunctzombie** [comment]
@jxson you worked on nesting support most recently. Thoughts about this?

**Comment by jxson** [comment]
hmm, yeah. The nesting support I added was because I was doing the same exact thing @spudly was with the qs module. I think the expectation for name attributes with numbered indexes should do what @spudly initially pointed out.

I am not sure something like this be serialized to though...

    <form>
      <input name="people[][name]" value="fred">
      <input name="people[][age]" value="12">

      <input name="people[][name]" value="bob">
      <input name="people[][age]" value="14">

      <input name="people[][name]" value="bubba">
      <input name="people[][age]" value="15">
    </form>

**Comment by defunctzombie** [comment]
ping on this issue. Would entertain a PR to look into this if still desired.

**Comment by jxson** [comment]
I can take a crack at it, might take a bit since I am traveling...

**Comment by jxson** [comment]
Just a heads up on this: I have a solid fix for this issue and #23 just cleaning up code and adding more tests before sending a PR.
